web/src/main/java/org/cbioportal/web/util/StudyViewFilterUtil.java

```diff
@@ -63,4 +78,77 @@ public ClinicalDataBin dataBinToClinicalDataBin(ClinicalDataBinFilter attribute,
         }));
     }
+
+    public Integer getFilteredCountByDataEquality(List<ClinicalDataFilter> attributes, MultiKeyMap clinicalDataMap,
+            String entityId, String studyId, Boolean negateFilters) {
+        Integer count = 0;
+        for (ClinicalDataFilter s : attributes) {
+            List<String> filteredValues = s.getValues().stream().map(DataFilterValue::getValue)
+                    .collect(Collectors.toList());
+            filteredValues.replaceAll(String::toUpperCase);
+            if (clinicalDataMap.containsKey(studyId, entityId, s.getAttributeId())) {
+                String value = (String) clinicalDataMap.get(studyId, entityId, s.getAttributeId());
+                if (negateFilters ^ filteredValues.contains(value)) {
+                    count++;
+                }
+            } else if (negateFilters ^ filteredValues.contains("NA")) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    public List<ClinicalDataCountItem> getClinicalDataCountsFromCustomData(List<CustomDataSession> customDataSessions,
+            Map<String, SampleIdentifier> filteredSamplesMap, List<Patient> patients) {
+        int totalSamplesCount = filteredSamplesMap.keySet().size();
+        int totalPatientsCount = patients.size();
+
+        return customDataSessions.stream().map(customDataSession -> {
+
+            Map<String, List<CustomDataValue>> groupedDatabyValue = customDataSession.getData().getData().stream()
+                    .filter(datum -> {
+                        return filteredSamplesMap
+                                .containsKey(getCaseUniqueKey(datum.getStudyId(), datum.getSampleId()));
+                    }).collect(Collectors.groupingBy(CustomDataValue::getValue));
+
+            ClinicalDataCountItem clinicalDataCountItem = new ClinicalDataCountItem();
+            clinicalDataCountItem.setAttributeId(customDataSession.getId());
+
+            List<ClinicalDataCount> clinicalDataCounts = groupedDatabyValue.entrySet().stream()
+                    .map(entry -> {
+                        long count = entry.getValue().stream().map(datum -> {
+                            return getCaseUniqueKey(datum.getStudyId(),
+                                    customDataSession.getData().getPatientAttribute()
+                                            ? datum.getPatientId()
+                                            : datum.getSampleId());
+        
+                        }).distinct().count();
+                        ClinicalDataCount dataCount = new ClinicalDataCount();
+                        dataCount.setValue(entry.getKey());
+                        dataCount.setCount(Math.toIntExact(count));
+                        return dataCount;
+                    })
+                    .filter(c -> !c.getValue().toUpperCase().equals("NA") && !c.getValue().toUpperCase().equals("NAN")
```

Comment:
Kinda makes me nervous burying these missing value constants in code like this and not putting them in a central place.

Reply:
Agreed. they are at many place all over the repo. cleanup should be in another pr